src/algorithms/xiao2013_utils.jl

```diff
@@ -0,0 +1,473 @@
+CountingMIS(3, 3)
+
+julia> max(CountingMIS(1, 1), CountingMIS(2, 2))
+CountingMIS(2, 3)
+```
+"""
+struct CountingMIS
+    mis::Int
+    count::Int
+    CountingMIS(mis::Int) = new(mis, 1)
+    CountingMIS(mis::Int, count::Int) = new(mis, count)
+end
+
+Base.:+(x::CountingMIS, y::Int) = CountingMIS(x.mis + y, x.count)
+Base.:+(x::Int, y::CountingMIS) = CountingMIS(x + y.mis, y.count)
+Base.:+(x::CountingMIS, y::CountingMIS) = CountingMIS(x.mis + y.mis, x.count + y.count)
+
+Base.max(x::CountingMIS, y::CountingMIS) = CountingMIS(max(x.mis, y.mis), x.count + y.count)
+Base.max(args...) = reduce(max, args)
+
+struct Path
+    vertices::Vector{Int}
+end
+
+add_left(path::Path, v::Int) = Path([v; path.vertices])
+add_right(path::Path, v::Int) = Path([path.vertices; v])
+left_boundary(path::Path) = path.vertices[1]
+right_boundary(path::Path) = path.vertices[end]
+left_neighbors(path::Path, g::SimpleGraph) = setdiff(neighbors(g, left_boundary(path)), path.vertices)
+right_neighbors(path::Path, g::SimpleGraph) = setdiff(neighbors(g, right_boundary(path)), path.vertices)
+Base.length(path::Path) = length(path.vertices)
+is_o_path(path::Path) = isodd(length(path.vertices))
+
+function is_path(g::SimpleGraph, path::Path)
+    for i in 2:length(path.vertices) - 1
+        degree(g, path.vertices[i]) != 2 && return false
+        Set(neighbors(g, path.vertices[i])) == Set([path.vertices[i-1], path.vertices[i+1]]) || return false
+    end
+    return true
+end
+
+
+const Beineke_graphs = [
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4), (2, 3), (2, 5), (3, 5), (4, 5)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 4), (1, 5), (2, 3), (2, 4), (2, 5), (3, 4), (3, 5), (4, 5)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (2, 3), (2, 4), (3, 4), (3, 5), (4, 5), (5, 6)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4), (2, 3), (2, 4), (2, 5), (3, 4), (4, 5), (5, 6)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4), (2, 3), (2, 4), (3, 4), (3, 5), (3, 6), (4, 5), (4, 6), (5, 6)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4), (2, 3), (2, 5), (3, 4), (4, 6), (5, 6)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (2, 3), (2, 4), (3, 4), (3, 5), (4, 5), (4, 6), (5, 6)])),
+    SimpleGraph(Graphs.SimpleEdge.([(1, 2), (1, 3), (1, 4), (2, 3), (2, 5), (3, 4), (3, 5), (3, 6), (4, 6), (5, 6)])),
+]
+
+function is_line_graph(g::SimpleGraph)
+    for check_graph in Beineke_graphs
+        if Graphs.Experimental.has_induced_subgraphisomorph(g,check_graph)
+            return false
+        end
+    end
+    return true
+end
+
+function remove_vertex(g, v)
+    g, vs = induced_subgraph(g, setdiff(vertices(g), v))
+    return g
+end
+
+function remove_vertices(g, vertices)
+    gc = copy(g)
+    rem_vertices!(gc, vertices)
+    return gc
+end
+
+function closed_neighbors(g::SimpleGraph, vertices::Vector{Int})

```

Comment:
These tow function will make you life much easier, far more better that `vcat(...)`